tests/contrib/test_gce.py

```diff
@@ -77,46 +92,50 @@ def test_refresh_token_failed_fetch(self):
             json.dumps({'access_token': 'a', 'expires_in': 100})
         )
         credentials = AppAssertionCredentials()
-
+        credentials.invalid = False
+        credentials.service_account_email = 'a@example.com'
         with self.assertRaises(HttpAccessTokenRefreshError):
-            credentials._refresh(http_request=http_request)
+            credentials._refresh(http_request)
 
     def test_serialization_data(self):
         credentials = AppAssertionCredentials()
         self.assertRaises(NotImplementedError, getattr,
                           credentials, 'serialization_data')
 
-    def test_create_scoped_required_without_scopes(self):
+    def test_create_scoped_required(self):
         credentials = AppAssertionCredentials()
         self.assertFalse(credentials.create_scoped_required())
 
-    @mock.patch('warnings.warn')
-    def test_create_scoped_required_with_scopes(self, warn_mock):
-        credentials = AppAssertionCredentials(['dummy_scope'])
-        self.assertFalse(credentials.create_scoped_required())
-        warn_mock.assert_called_once_with(_SCOPES_WARNING)
-
-    @mock.patch('warnings.warn')
-    def test_create_scoped(self, warn_mock):
-        credentials = AppAssertionCredentials()
-        new_credentials = credentials.create_scoped(['dummy_scope'])
-        self.assertNotEqual(credentials, new_credentials)
-        self.assertTrue(isinstance(new_credentials, AppAssertionCredentials))
-        self.assertEqual('dummy_scope', new_credentials.scope)
-        warn_mock.assert_called_once_with(_SCOPES_WARNING)
-
     def test_sign_blob_not_implemented(self):
         credentials = AppAssertionCredentials([])
         with self.assertRaises(NotImplementedError):
             credentials.sign_blob(b'blob')
 
     @mock.patch('oauth2client.contrib._metadata.get_service_account_info',
-                return_value={'email': 'a@example.com'})
-    def test_service_account_email(self, metadata):
+                return_value=SERVICE_ACCOUNT_INFO)
+    def test_retrieve_scopes(self, metadata):
```

Comment:
Testing the externally surfaced function `gce.AppAssertionCredentials.retrieve_scopes` seems preferable to testing an class private function?